src/radical/pilot/agent/executing/popen.py

```diff
@@ -52,17 +52,18 @@ class Popen(AgentExecutingComponent):
     _header    = '#!/bin/sh\n'
     _separator = '\n# ' + '-' * 78 + '\n'
 
+    # flags for the watcher queue
+    TO_WATCH  = 0
+    TO_CANCEL = 1
+
     # --------------------------------------------------------------------------
     #
     def initialize(self):
 
       # self._log.debug('popen initialize start')
         AgentExecutingComponent.initialize(self)
 
-        self._cancel_lock     = mt.RLock()
-        self._tasks_to_cancel = list()
-        self._tasks_to_watch  = list()
-        self._watch_queue     = queue.Queue()
+        self._watch_queue = queue.Queue()
```

Comment:
maybe to have two queues instead? (`_watch_queue`, `_cancel_queue`), just wondering about possibility to have task cancelation delayed, could it be?